server/util/authutil/authutil.go

```diff
@@ -30,3 +30,13 @@ func AuthorizeGroupRole(u interfaces.UserInfo, groupID string, allowedRoles role
 	}
 	return nil
 }
+
+// IsAnonymousUserError can be used to check whether an error returned by
+// functions which return the authenticated user (such as AuthenticatedUser or
+// AuthenticateSelectedGroupID) is due to an anonymous user accessing the
+// service. This is useful for allowing anonymous users to proceed, in cases
+// where anonymous usage is explicitly enabled in the app config, and we support
+// anonymous usage for the part of the service where this is used.
+func IsAnonymousUserError(err error) bool {
```

Comment:
happy to also rename this to `IsMissingCredentialsError` as part of this PR if people think it's clearer.

Reply:
Will follow up in a separate PR with this, if we want to do it.